### PR TITLE
feat(LAB-3244): on LLM dynamic projects export annotations at convers…

### PR DESCRIPTION
**Pull request by FannyGaudin**
…ation level

### CODE REVIEW

File: src/kili/llm/services/export/dynamic.py [hunk 1]
```diff
@@ -192,8 +195,8 @@ def _format_comparison_annotation(annotation, completions, job, obfuscated_model
 
 def _format_json_response(
     jobs_config: Dict, annotations: List[Dict], completions: List[Dict], obfuscated_models: Dict
-) -> Dict[str, Union[str, List[str]]]:
-    result = {}
+) -> Dict[str, Dict]:
```
Comment (HNicolas) [inline src/kili/llm/services/export/dynamic.py:198]: Don't you loose some of the typing here ?

File: src/kili/llm/services/export/dynamic.py [hunk 2]
```diff
@@ -60,25 +61,28 @@ def export(
...
+                        obfuscated_models,
+                    )
+                    label_data = {
+                        "author": label["author"]["email"],
+                        "created_at": label["createdAt"],
+                        "label_type": label["labelType"],
+                        "label": formatted_response["label"],
+                    }
+                    if step == total_rounds - 1 and formatted_response["conversation_label"]:
+                        label_data["conversation_label"] = formatted_response["conversation_label"]
```
Comment (baptiste-olivier) [inline src/kili/llm/services/export/dynamic.py:77]: I discussed with Stéphanie about that,  if we want to differentiate the type of the annotation at the label level, we need to do it for every levels ; for example : 

```
...
    "label": {
            "completion": {
                "RANKING": [
                    "B_2"
                ]
            },
            "conversation": {
                "TRANSCRIPTION_1": "some transcription at asset level"
            },
            "chat_item": {
                "012399314932R4": {
                    "TRANSCRIPTION_2": "some transcription at chat item level"
                }
            }
        }
```

Comment (baptiste-olivier) [reply src/kili/llm/services/export/dynamic.py:77]: I'm not really ok with this format, as we duplicate some information already present in the json_interface. We can discuss it with the team if you want

Comment (baptiste-olivier) [reply src/kili/llm/services/export/dynamic.py:77]: And doing this change mean applying it also on static export for consistency,  creating a breaking change in our export format

File: src/kili/llm/services/export/dynamic.py (hunk 1 above)
Comment (FannyGaudin) [reply src/kili/llm/services/export/dynamic.py:198]: Yes I will update the type

File: src/kili/llm/services/export/dynamic.py (hunk 2 above)
Comment (FannyGaudin) [reply src/kili/llm/services/export/dynamic.py:77]: No problem but could you tell me what format you have in mind ?

File: src/kili/llm/services/export/dynamic.py [hunk 3]
```diff
@@ -168,6 +172,7 @@ def _build_rounds(self, chat_items, annotations, json_interface):
                         annotation
                         for annotation in annotations
                         if annotation["chatItemId"] == chat_item["id"]
+                        or annotation["chatItemId"] is None
```
Comment (baptiste-olivier) [inline src/kili/llm/services/export/dynamic.py:175]: Not sure we should add this condition here.

If we do so, for each assistant response we will add all annotations without chatItemId.
If our asset haven't any assistant response, annotation whithout chat item id won't be added.

For me we should do it separately, just before appending the current_round to rounds.

File: src/kili/llm/services/export/dynamic.py
Comment (baptiste-olivier) [inline src/kili/llm/services/export/dynamic.py:1]: We should modify static export as well, to keep it in sync with dynamic export

File: src/kili/llm/services/export/dynamic.py [hunk 4]
```diff
@@ -59,25 +60,30 @@ def export(
...
+                    )
+                    label_data = {
+                        "author": label["author"]["email"],
+                        "created_at": label["createdAt"],
+                        "label_type": label["labelType"],
+                        "label": {},
+                    }
+                    if formatted_response["turn"]:
+                        label_data["label"]["turn"] = formatted_response["turn"]
+                    if step == total_rounds - 1 and formatted_response["asset"]:
```
Comment (baptiste-olivier) [inline src/kili/llm/services/export/dynamic.py:77]: Did you double check with @stephnguyenkili if we wanted this information only for the last round?

If the main purpose for the customer is to handle each round as a separate result, we may need to export it on each rounds

File: src/kili/llm/services/export/dynamic.py [hunk 5]
```diff
@@ -173,6 +179,9 @@ def _build_rounds(self, chat_items, annotations, json_interface):
                         has_children = True
                         parent_target = chat_item["id"]
 
+                current_round["annotations"] += [
```
Comment (baptiste-olivier) [inline src/kili/llm/services/export/dynamic.py:182]: we'll need to do this if we have a pending round when the loop finish (l203, before the ` rounds.append(current_round)`)

File: src/kili/llm/services/export/dynamic.py (hunk 4 above)
Comment (FannyGaudin) [reply src/kili/llm/services/export/dynamic.py]: Check with @stephnguyenkili that's ok tu put on the last round

File: src/kili/llm/services/export/dynamic.py (hunk 5 above)
Comment (FannyGaudin) [reply src/kili/llm/services/export/dynamic.py:182]: Change it thanks

File: src/kili/llm/services/export/static.py [hunk 6]
```diff
@@ -83,6 +83,8 @@ def _format_json_response(
             result[job_name] = job_value["text"]
         else:
             logging.warning(f"Job {job_name} with mlTask {job_config['mlTask']} not supported")
+    if len(result) != 0:
+        return {"asset": result}
```
Comment (baptiste-olivier) [inline src/kili/llm/services/export/static.py:87]: To be consistent, this should be `conversation` as well

Comment (FannyGaudin) [reply src/kili/llm/services/export/static.py]: yes ^^

**Review by HNicolas — APPROVED** [review]

**Pull request merged**